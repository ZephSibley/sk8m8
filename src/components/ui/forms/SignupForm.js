import React from 'react';
import Button from '@material-ui/core/Button'

import formStyles from '../../../styles/forms';
import ValidInput from '../ValidInput';

const SignupForm = () => {
    const formClasses = formStyles();

    const signupHandler = async () => {
        //await AsyncStorage.setItem('userToken', 'abc');
        console.log('signup!')
    };

    return (
        <form className={formClasses.form}>
            <ValidInput
                label='email'
                validationrules={{
                    isEmail: true,
                }}
            />
            <ValidInput
                label='confirm email'
                validationrules={{
                    equalTo: 'email', // TODO: configure equalTo
                }}
                controlvalue={''}
            />
            <ValidInput
                label='username'
                validationrules={{
                    minLength: 3,
                }}
            />
            <ValidInput
                label='password'
                validationrules={{
                    minLength: 6
                }}
            />
            <ValidInput
                label='confirm password'
                validationrules={{
                    equalTo: 'password'
                }}
                controlvalue={''}
            />
            <input
                style={{ display: 'none' }}
                id="outlined-button-submit"
                multiple
                type="submit"
            />
            <label htmlFor="outlined-button-file">
                <Button
                    variant="outlined"
                    color="primary"
                    component="span"
                    className={formClasses.submit}
                    onClick={(e) => signupHandler(e)}
                >
                    Sign up
              </Button>
            </label>
        </form>
    )
}

export default SignupForm;