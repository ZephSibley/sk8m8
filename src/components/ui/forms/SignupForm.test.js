import React from 'react';
import renderer from 'react-test-renderer';
import SignupForm from './SignupForm';

test('Basic Rendering', () => {
    const component = renderer.create(
        <SignupForm />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})