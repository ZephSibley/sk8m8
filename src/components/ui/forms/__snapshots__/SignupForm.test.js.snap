// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div>
  <form
    className="makeStyles-form-1"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        email
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="email"
          value=""
        />
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        confirm email
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="confirm_email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="email"
          value=""
        />
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        username
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="username"
          value=""
        />
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        password
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        confirm password
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="confirm_password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
      </div>
      
    </div>
    <span
      className="MuiFormHelperText-root Mui-error"
    />
    <label>
      <input
        name="terms"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
           terms and conditions.
        </span>
      </button>
    </label>
    <button
      className="MuiButtonBase-root MuiButton-root makeStyles-submit-5 MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
      disabled={true}
      margin="normal"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Sign up
      </span>
    </button>
  </form>
</div>
`;
